### ISSUE TITLE
hitting 'n' or searching for anything at '(END)' exits in v670

### ISSUE THREAD

**Issue by smoser** [gwsw/less#592]
This seems to be a regression to me between v669 and v670.

If I page to end of less input and then search for something version 670 will exit.  version 690 would not.
It may seem like a strange user behavior, but I often will hit space multiple times to view content and then at type '/some-string<cr>', and then hit 'N' to go back to the string above.

I can reproduce like this:
```
$ for n in $(seq 1 50); do echo "$n"; done > my-file.txt
$ less < my-file.txt
```

my-file.txt has 50 lines, I hit space bar once, and then type '/23' and less will exit.

less will stay open if either 
a.) version 669 is used. 
b.) input is from a file on the command line rather than stdin `less my-file.txt`

**Comment by gwsw** [comment]
I think this was fixed in 8db7960040647b94812309d4b48ea9614bd04c55. Let me know if you see it in that or any later version.

**Comment by smoser** [comment]
> I think this was fixed in https://github.com/gwsw/less/commit/8db7960040647b94812309d4b48ea9614bd04c55. Let me know if you see it in that or any later version.

I cherry-picked that commit and it seems to be fixed.
thanks for quick  reply.
